zomes/chat/src/validation.rs

```diff
@@ -110,27 +118,29 @@ pub(crate) fn common_validatation(data: ValidateData) -> ExternResult<ValidateCa
         _ => return Ok(ValidateCallbackResult::Valid),
     };
     if let Entry::Agent(_) = entry {
-        match data.element.header().prev_header() {
-            Some(header) => {
-                match get(header.clone(), GetOptions::default()) {
-                    Ok(element_pkg) => match element_pkg {
-                        Some(element_pkg) => {
-                            match element_pkg.signed_header().header() {
-                                Header::AgentValidationPkg(pkg) => {
-                                    return joining_code(pkg.author.clone(), pkg.membrane_proof.clone(), false)
+        if !skip_proof() {
+            match data.element.header().prev_header() {
+                Some(header) => {
+                    match get(header.clone(), GetOptions::default()) {
+                        Ok(element_pkg) => match element_pkg {
+                            Some(element_pkg) => {
+                                match element_pkg.signed_header().header() {
+                                    Header::AgentValidationPkg(pkg) => {
+                                        return joining_code(pkg.author.clone(), pkg.membrane_proof.clone(), false)
+                                    }
+                                    _ => return Ok(ValidateCallbackResult::Invalid("No Agent Validation Pkg found".to_string()))
                                 }
-                                _ => return Ok(ValidateCallbackResult::Invalid("No Agent Validation Pkg found".to_string()))
-                            }
+                            },
+                            None => return Ok(ValidateCallbackResult::UnresolvedDependencies(vec![(header.clone()).into()]))
                         },
-                        None => return Ok(ValidateCallbackResult::UnresolvedDependencies(vec![(header.clone()).into()]))
-                    },
-                    Err(e) => {
-                        debug!("Error on get when validating agent entry: {:?}; treating as unresolved dependency",e);
-                        return Ok(ValidateCallbackResult::UnresolvedDependencies(vec![(header.clone()).into()]))
+                        Err(e) => {
+                            debug!("Error on get when validating agent entry: {:?}; treating as unresolved dependency",e);
+                            return Ok(ValidateCallbackResult::UnresolvedDependencies(vec![(header.clone()).into()]))
+                        }
                     }
-                }
-            },
-            None => return Ok(ValidateCallbackResult::Invalid("Impossible state".to_string()))
+                },
+                None => return Ok(ValidateCallbackResult::Invalid("Impossible state".to_string()))
```

Comment:
Even though this is definitely impossible, it would be more appropriate to `panic!` here, which would return an error to Holochain, and the app would be deactivated. Mistakes in app logic should not punish users with warrants wherever possible.
```suggestion
                None => unreachable!("This element will always have a prev_header")
```